src/Analyzer.PowerShellRuleEngine/PowerShellRuleEngine.cs

```diff
@@ -102,108 +45,51 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(TemplateContext templateContext)
                 throw new ArgumentException($"{nameof(TemplateContext.TemplateIdentifier)} must not be null.", nameof(templateContext));
             }
 
-            if (runspace == null)
+            if (templateContext?.ExpandedTemplate == null)
             {
-                // There was an error loading the TTK module.  Return an empty collection.
-                logger?.LogWarning("Unable to run PowerShell based checks.  Initialization failed.");
-                return Enumerable.Empty<IEvaluation>();
+                throw new ArgumentException($"{nameof(TemplateContext.ExpandedTemplate)} must not be null.", nameof(templateContext));
             }
 
-            var executionResults = Powershell.Create(runspace)
-                .AddCommand("Test-AzTemplate")
-                .AddParameter("Test", "deploymentTemplate")
-                .AddParameter("TemplatePath", templateContext.TemplateIdentifier)
-                .Invoke();
-
-            var evaluations = new List<PowerShellRuleEvaluation>();
-
-            foreach (dynamic executionResult in executionResults)
+            if (templateContext?.ResourceMappings == null)
             {
-                var uniqueErrors = new Dictionary<string, SortedSet<int>>(); // Maps error messages to a sorted set of line numbers
-
-                foreach (dynamic warning in executionResult.Warnings)
-                {
-                    PreProcessErrors(warning, uniqueErrors);
-                }
-
-                foreach (dynamic error in executionResult.Errors)
-                {
-                    PreProcessErrors(error, uniqueErrors);
-                }
-
-                foreach (KeyValuePair<string, SortedSet<int>> uniqueError in uniqueErrors)
-                {
-                    var ruleId = (executionResult.Name as string)?.Replace(" ", "");
-                    ruleId = !String.IsNullOrEmpty(ruleId) ? ruleId : "TTK";
-                    var ruleDescription = executionResult.Name + ". " + uniqueError.Key;
-
-                    foreach (int lineNumber in uniqueError.Value)
-                    {
-                        evaluations.Add(new PowerShellRuleEvaluation(ruleId, ruleDescription, false, new PowerShellRuleResult(false, lineNumber)));
-                    }
-                }
+                throw new ArgumentException($"{nameof(TemplateContext.ResourceMappings)} must not be null.", nameof(templateContext));
             }
 
-            return evaluations;
-        }
-
-        private void PreProcessErrors(dynamic error, Dictionary<string, SortedSet<int>> uniqueErrors)
-        {
-            var lineNumber = 0;
+            // TODO: Temporary work-around: write template to disk so PSRule will analyze it
```

Comment:
Don't have context - why is this work-around needed/what is the future plan?

Reply:
Because PSRule needs to read the template from a file to do some of the checks. The future plan is to make the needed fixes in there so that we can just pass the template string